src/ops/constant.rs

```diff
@@ -264,17 +290,8 @@ mod test {
             classic_row![ClassicType::i64(), ClassicType::F64],
             type_row![],
         ];
-        let pred_ty = SimpleType::new_predicate(pred_rows.clone());
 
-        let mut b = DFGBuilder::new(type_row![], SimpleRow::from(vec![pred_ty])).unwrap();
-        let res = b.add_constant(ConstValue::predicate(
-            0,
-            ConstValue::Tuple(vec![]),
-            pred_rows,
-        ));
-        assert_eq!(
-            res,
-            Err(BuildError::BadConstant(ConstTypeError::TupleWrongLength))
-        );
+        let res = Const::predicate(0, ConstValue::Tuple(vec![]), pred_rows);
+        assert_matches!(res, Err(ConstTypeError::TupleWrongLength));
```

Comment:
Hehe - so we never get as far as calling `add_constant`. Nice. I think you can remove BuildError::BadConstant, thus.

Reply:
I still think it's useful as a BuildError (it alllows use of `?` inside builder
contexts on `Const::new`)